client/src/components/Form/Elements/FormSelect.vue

```diff
@@ -0,0 +1,188 @@
+            <b-col v-else>
+                <!-- Multiple select from drop down -->
+                <multiselect
+                    v-if="multiple"
+                    v-model="currentValue"
+                    :options="formattedOptions"
+                    :multiple="true"
+                    placeholder="Select value"
+                    deselect-label=""
+                    select-label=""
+                    track-by="value"
+                    label="label">
+                </multiselect>
+                <!-- Single select from drop down -->
+                <multiselect
+                    v-else
+                    v-model="currentValue"
+                    :options="formattedOptions"
+                    deselect-label=""
+                    select-label=""
+                    label="label"
+                    track-by="value">
+                    {{ currentValue }}
+                </multiselect>
+            </b-col>
+        </b-row>
+    </div>
+</template>
+
+<script>
+import Multiselect from "vue-multiselect";
+
+export default {
+    components: {
+        Multiselect,
+    },
+    props: {
+        value: {
+            default: null,
+        },
+        defaultValue: {
+            type: [String, Array],
+            required: true,
+        },
+        options: {
+            type: Array,
+            required: true,
+        },
+        multiple: {
+            type: Boolean,
+            required: true,
+            default: false,
+        },
+        display: {
+            type: String,
+            required: false,
+            default: null,
+        },
+        optional: {
+            type: Boolean,
+            required: false,
+            default: null,
+        },
+    },
+    computed: {
+        formattedOptions() {
+            const formattedOptions = [];
+            this.options.map((option, i) => {
+                formattedOptions[i] = {
+                    label: option[0],
+                    value: option[1],
+                    default: option[2],
+                };
+            });
+            if (!this.multiple && this.optional) {
+                formattedOptions.unshift({
+                    label: "Nothing selected",
+                    value: null,
+                    default: false,
+                });
+            }
+            return formattedOptions;
+        },
+        currentValue: {
+            get() {
+                // My understanding of this is that if we start with a value prop, use it.
+                // If there's a defaultValue set, find it in the array and use that.
+                // If there's no default value set, use the first item in the array with 'default' set to true.
+                // Lastly, just default to the first element in the array.
+                if (this.value == null) {
+                    return;
+                } else if (this.value !== "") {
+                    if (this.multiple && this.display != "checkboxes") {
```

Comment:
Strict equality can be used in several places here.

```suggestion
                    if (this.multiple && this.display !== "checkboxes") {
```

Reply:
Why are we using strict equality here?

Reply:
It is good practice to treat strict equality as the default for equality checking. Despite being the longer operator, strict equality is simpler and faster than `==` equality. The reason for this is that the "old" equality check (`==`) performs implicit type conversion. If this conversion is not needed, or not wanted, a simple equality check is sufficient and preferable.

Reply:
Yes there are definitely benefits, but it's not simpler. It adds an additional character to every comparison, possibly across the entire client code-base. So far we commonly use `==`, and strict equality only when implicit conversion should be explicitly avoided, usually in cases comparing with `null` and `undefined`. Implicit conversion is also likely to be extremely fast, although it is often avoided, like in the above example, in which `display` is a typed incoming `String` prop. Also TypeScript will help. Having said this, this is an interesting topic, should maybe have its own issue.

Reply:
Not simpler as in shorter, simpler as in easier to understand and reason with. Just compare the description for these operators on mdn:

`===` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality#description
`==` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Equality#description